pkg/plugin/v2_server.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package plugin
+
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/kubernetes-kms/pkg/config"
+	"github.com/Azure/kubernetes-kms/pkg/metrics"
+
+	"k8s.io/klog/v2"
+	kmsv2 "k8s.io/kms/apis/v2"
+)
+
+// KeyManagementServiceV2Server is a gRPC server.
+type KeyManagementServiceV2Server struct {
```

Comment:
Can we use the higher level logic in https://github.com/kubernetes/kms/blob/release-1.27/pkg/service/grpc_service.go ?

Reply:
We could have but in the plugin, we have wiring for metrics recording, so I am keeping this implementation. Let me know your thoughts